### ISSUE TITLE
Bug: EventSystem inconsistency in emitted vs listened event types

### ISSUE THREAD

**Issue by luxaritas** [pixijs/pixi.js#9035]
### Current Behavior

When listening to `FederatedEvent`s that have multiple versions based on pointer types, the `FederatedEvent#type` property doesn't match the listened event type

### Expected Behavior

When listening to `FederatedEvent`s that have multiple versions based on pointer types, the `FederatedEvent#type` property is the same as the listened event type

### Steps to Reproduce

https://www.pixiplayground.com/#/edit/rh_8gwtaysmTiltMY2OC4

### Environment

Tested the playground link with v7.1.0-alpha on FF 108.0.1, however initially encountered this in attempting to take the partial step of migrating my v6.5.8 app to EventSystem

### Additional Information

I'm not completely sure what the ideal solution is here. It seems to me like the "correct" behavior should be always using the same type as the original event. At the very least, it's quite surprising that most events use the `pointer` variant (eg `pointerup` instead of `mouseup`, however `click` is used instead of `pointertap`). However, this is significant enough that it could incur a breaking change if applications are filtering events based on the current scheme.

**Comment by miltoncandelero** [comment]
Pointer is the blanket term for "any pointing device" it is good to merge touch and mouse 🤔

**Comment by Zyie** [comment]
It seems like this was a explicit choice for events

https://github.com/pixijs/pixijs/blob/79bb8aa5bcdb0f977b3246d63d7dd1fd71bf8cb5/packages/events/src/EventSystem.ts#L650

Although I'm not sure if it was the correct one. Seems like it can be both useful and not useful depending on the circumstance

Feels like maybe we shouldn't mess with the type?

@bigtimebuddy do you have any thoughts?

**Comment by bigtimebuddy** [comment]
@ShukantPal do you have any context to offer here?

I'm not a big fan of transforming the type.

**Comment by miltoncandelero** [comment]
I think we might have a confusion between `type` and `pointerType` 🤔 

https://www.pixiplayground.com/#/edit/mgXhcnF8NVOfGUzJjNdCD

**Comment by ShukantPal** [comment]
I think the idea was that we're emulating a `PointerEvent` in `FederatedPointerEvent`.

If we want specifically `mouse` and `touch` events, then we'd create implementations of `FederatedMouseEvent` and `FederatedTouchEvent` to match DOM behavior. For example, [TouchEvent](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent) doesn't actually inherit from [Pointer Event](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent).

**Comment by luxaritas** [comment]
For what it's worth: I'm _personally_ fine with always using pointer types since I only use pointer events - however if that's the case, the `click` type should be translated to pointerTap. If you want to fully emulate DOM events, then the types shouldn't be transformed and class implementations should match the DOM. Otherwise it might even make sense to _only_ support pointer events.

**Comment by ShukantPal** [comment]
Makes sense!, and thanks for finding these gaps in the initial implementation.
